app/main/views/g_cloud.py

```diff
@@ -420,7 +420,10 @@ def save_search(framework_family):
     framework = framework_helpers.get_latest_live_framework_or_404(all_frameworks, framework_family)
     lots_by_slug = framework_helpers.get_lots_by_slug(framework)
 
-    search_query = url_decode(request.values.get('search_query'))
+    try:
+        search_query = url_decode(request.values['search_query'])
+    except KeyError:
+        abort(400)
```

Comment:
Do we need a `try...except`? How about

```
    if 'search_query' not in request.values:
        abort(400)
```

Reply:
Sure that would work. Out of curiosity, what is the reason to avoid the `try...except`?

Reply:
I've made the change.

Reply:
> Out of curiosity, what is the reason to avoid the try...except?

Good question! It's no less valid to do a `try...except`, it just feels easier to understand what's going on from the `if` statement?